PYME/cluster/clusterUI/clusterbrowser/views.py

```diff
@@ -110,7 +110,7 @@ def upload(request, directory):
     request.upload_handlers.insert(0, TemporaryFileUploadHandler(request))
     return upload_files(request, directory)
 
-@csrf_protect
+@csrf_exempt
```

Comment:
This is definitely not something we should be exempting (the fact that we had specifically asked to protect it might be a clue). In practice we should probably protect this endpoint better - i.e. enforce authentication once that is fully rolled out.

Reply:
re:clues `upload_file` just above it is exempt